test/samlRequest.spec.ts

```diff
@@ -0,0 +1,127 @@
+import { SAML } from "../src/saml";
+import { FAKE_CERT, SamlCheck } from "./types";
+import * as zlib from "zlib";
+import * as should from "should";
+import { parseString } from "xml2js";
+
+const capturedSamlRequestChecks: SamlCheck[] = [
+  {
+    name: "Config with Extensions",
+    config: {
+      entryPoint: "https://wwwexampleIdp.com/saml",
+      cert: FAKE_CERT,
+      samlExtensions: {
+        "md:RequestedAttribute": {
+          "@isRequired": "true",
+          "@Name": "Lastname",
+          "@xmlns:md": "urn:vetuma:SAML:2.0:metadata",
```

Comment:
@kdhttps I think you meant to use namespace `urn:oasis:names:tc:SAML:2.0:metadata` for `md` prefix instead of `urn:vetuma:SAML:2.0:metadata` (because `urn:oasis:names:tc:SAML:2.0:metadata` would be inline with example you have written to README).

Reply:
Yes, but I found that [saml-schema-protocol-2.0.xsd](https://docs.oasis-open.org/security/saml/v2.0/saml-schema-protocol-2.0.xsd) not really define rules to validate specific namespace value to specific element. It just check that there should be namespace attribute with value. 
For Example: 
```
<samlp:Extensions xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol">
      <md:RequestedAttribute xmlns:md="urn:vetuma:SAML:2.0:metadata" isRequired="true" Name="Lastname" />
      <vetuma xmlns="urn:vetuma:SAML:2.0:extensions">
        <LG>sv</LG>
      </vetuma>
</samlp:Extensions>
```
I checked with above data with this [XML check Tool](https://www.samltool.com/validate_xml.php). It allows this namespace. Please let me know if there are rules to check specific namespace value for elements.

Reply:
Yes. Extensions content can be sequence of anything as long as it is well formed xml. My point was that since you choosed to select `RequestedAttribute` as one of the examples at the README.md and because `md:RequestedAttribute` happens to be defined in the specification that you referenced at README.md and because NS for `md:RequestedAttribute` happens to be `urn:oasis:names:tc:SAML:2.0:metadata` at that document you could keep using `RequestedAttribute` from the same namespace in the test case also.

Reply:
Oh Got it ! Just fixed. Now it is `urn:oasis:names:tc:SAML:2.0:metadata`. I though you want me to perform some validation on namespace value. Thank you!